src/cursor/cur_std.c

```diff
@@ -346,6 +389,10 @@ __wt_cursor_get_keyv(WT_CURSOR *cursor, uint32_t flags, va_list ap)
     if (!F_ISSET(cursor, WT_CURSTD_KEY_SET))
         WT_ERR(__wt_cursor_kv_not_set(cursor, true));
 
+    /* Force an allocated copy when using cursor copy debug. */
+    if (F_ISSET(S2C(session), WT_CONN_DEBUG_CURSOR_COPY))
+        WT_ERR(__wt_buf_grow(session, &cursor->key, cursor->key.size));
```

Comment:
I don't understand why this forces the buffer to grow, unless we know at this point that `cursor` doesn't point to already-allocated memory.  Why is that known to be the case?

Reply:
I'm not sure about your question.  We don't care about the buffer growing, only that the buffer `data` pointer puts to memory malloced by the buffer.  We call `__wt_buf_grow` because it has that side effect.  Put another way, if the `cursor` points to already-allocated memory, then that's okay (somebody `malloced` it in the past, so it will be tracked) and `__wt_buf_grow` will do nothing.  If `cursor` doesn't point to already-allocated memory, then `__wt_buf_grow` will make sure it does.

Reply:
By the time `__wt_cursor_get_valuev` (or `__wt_cursor_get_keyv`) is called, the data for the value is already in the WT_CURSOR->value.  The `get_valuevv` function just assigns the caller's pointers into point into the block of memory that is the total set of values.  So any new memory needs to allocated and data copied into place before that's done.